ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/LedgerApiBenchTool.scala

```diff
@@ -16,47 +23,67 @@ import java.util.concurrent.{
   ThreadPoolExecutor,
   TimeUnit,
 }
-import scala.concurrent.{Await, ExecutionContext, Future}
 import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, Future}
 
 object LedgerApiBenchTool {
   def main(args: Array[String]): Unit = {
     Cli.config(args) match {
       case Some(config) =>
         val benchmark = runBenchmark(config)(ExecutionContext.Implicits.global)
         Await.result(benchmark, atMost = Duration.Inf)
+        ()
       case _ =>
         logger.error("Invalid configuration arguments.")
     }
   }
 
   private def runBenchmark(config: Config)(implicit ec: ExecutionContext): Future[Unit] = {
-    logger.info(s"Starting benchmark")
-    logger.info(config.toString) //TODO: print a nice representation
+    val printer = pprint.PPrinter(200, 1000)
+    logger.info(s"Starting benchmark with configuration:\n${printer(config).toString()}")
 
     implicit val resourceContext: ResourceContext = ResourceContext(ec)
 
-    val channel = for {
+    val resources = for {
       executorService <- threadPoolExecutorOwner(config.concurrency)
       channel <- channelOwner(config.ledger, executorService)
-    } yield channel
+      system <- actorSystemResourceOwner()
+    } yield (channel, system)
 
-    channel.use { channel =>
-      Future {
-        val ledgerIdentityService: LedgerIdentityService = new LedgerIdentityService(channel)
-        val ledgerId: String = ledgerIdentityService.fetchLedgerId()
-        val transactionService = new TransactionService(channel, ledgerId)
-        config.streamConfig.foreach { streamConfig =>
+    resources.use { case (channel, system) =>
+      val ledgerIdentityService: LedgerIdentityService = new LedgerIdentityService(channel)
+      val ledgerId: String = ledgerIdentityService.fetchLedgerId()
+      val transactionService = new TransactionService(channel, ledgerId)
+      Future
+        .traverse(config.streams) { streamConfig =>
```

Comment:
The parallelism in this traverse is theoretically bounded by the ExecutionContext size, and could lead to confusing results if more streams than CPU cores are started

Reply:
Please correct me if I'm wrong, but having a separately configured thread pool for the gRPC channel (line 48) should allow us to run multiple transaction streams with requested concurrency parameters. The global execution context will be used just for spinning up all those streams.

Reply:
Oh, those services are using the configured thread pool.. Sometimes it's just better to review in the IDE! Apologies

Reply:
Oh, those services are using the configured thread pool.. Sometimes it's just better to review in the IDE! Apologies